### DISCUSSION TITLE
Character count limit for title field

### DISCUSSION THREAD

**Discussion by TheFunkyMonk** [craftcms/cms#11732] [Ideas] [answered]
I have a situation where we'd like to limit the character count for an entry's title field, like you can with custom plain text fields. Possible to add to the field preview window in the layout designer, maybe under an "Advanced" toggle if we want to keep the interface clean?

**Comment by brandonkelly** [comment, marked as answer]
We can definitely consider this down the road. For now, you can enforce it by registering a custom validation rule from a module:

```php
use craft\elements\Entry;
use craft\events\DefineRulesEvent;
use yii\base\Event;

Event::on(
    Entry::class,
    Entry::EVENT_DEFINE_RULES,
    function(DefineRulesEvent $event) {
        /** @var Entry $entry */
        $entry = $event->sender;

        if ($entry->section->handle === 'mySectionHandle') {
            $event->rules[] = ['title', 'string', 'max' => 50];
        }
    }
);
```
